src/telemetry/reqwest_middleware.rs

```diff
@@ -0,0 +1,35 @@
+use http::Extensions;
+use opentelemetry_http::HeaderInjector;
+use reqwest::{Request, Response};
+use reqwest_middleware::{Middleware, Next, Result};
+use tracing::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt as _;
+
+/// Middleware for [reqwest-middleware] to propagate the Otel context
+///
+/// Example
+///
+/// let reqwest_client = reqwest::Client::builder().build().unwrap();
+///	let client = ClientBuilder::new(reqwest_client)
+///    // Insert the tracing middleware
+///    .with(OtelMiddleware::default())
+///    .build();
+/// client.get("http://localhost").send().await;
```

Comment:
```suggestion
/// ```
/// let reqwest_client = reqwest::Client::builder().build().unwrap();
///	let client = ClientBuilder::new(reqwest_client)
///    // Insert the tracing middleware
///    .with(OtelMiddleware::default())
///    .build();
/// client.get("http://localhost").send().await;
/// ```
```